src/lib/components/toggle/QSwitch.scss

```diff
@@ -0,0 +1,267 @@
+@use "sass:list";
+@use "$css/mixins";
+
+$root: ".q-switch";
+$track: "#{$root}__track";
+$handle: "#{$root}__handle";
+$icon: "#{$root}__icon";
+
+$checked: "[aria-checked]";
+$unchecked: ":not([aria-checked])";
+
+$check-map: (
+  "checked": "[aria-checked]",
+  "unchecked": ":not([aria-checked])",
+);
+
+@function getColor($el, $checked: false, $hovered: null) {
+  @if $el == "track" {
+    @return if($checked, var(--primary), var(--surface-container-highest));
+  } @else if $el == "icon" {
+    @return if($checked, var(--on-primary-container), var(--surface-container-highest));
+  } @else if $el == "handle" {
+    @if ($checked) {
+      @return if($hovered, var(--primary-container), var(--on-primary));
+    } @else {
+      @return if($hovered, var(--on-surface-variant), var(--outline));
+    }
+  } @else {
+    @error "The given element #{$el} is not supported.";
+  }
+}
+
+.q-switch {
+  position: relative;
+
+  display: inline-flex;
+  align-items: center;
+  flex-shrink: 0; // Stop from collapsing in flex containers
```

Comment:
No need to change this now, but comments should be at the top of the line, not right to it.